source/simulator/stokes_matrix_free.cc

```diff
@@ -1848,6 +1859,20 @@ namespace aspect
               for (unsigned int d=0; d<dim; ++d)
                 sym_grad_u[d][d] -= viscosity_x_2/3.0*div;
 
+            if (active_cell_data.enable_newton_derivatives)
+              {
+                SymmetricTensor<2,dim,VectorizedArray<double>> grads_phi_u_i = velocity.get_symmetric_gradient(q);
+
+                sym_grad_u +=  (grads_phi_u_i * active_cell_data.strain_rate_table(cell,q))
+                               * active_cell_data.newton_factor_wrt_strain_rate_table(cell,q);
```

Comment:
This is now no longer the symmetric gradient, i.e., the variable *name* no longer matches its *value*. I think it would be better to do
```
  if (enable_newton_derivatives)
  {
    const something = sym_grad_u + ...;
    velocity.submit_sym_grad(...);
  }
  else
    velocity.submit_sym_grad (existing arguments);
```

Reply:
> No difference in test output...

Exactly! Is it because in the Newton method, the rhs is already the residual, so the system already has homogeneous Dirichlet boundary conditions?